assimilation_code/programs/readme.rst

```diff
@@ -0,0 +1,206 @@
+   Prints out a quick table of obs types and counts, overall start and stop times, 
+   and metadata strings and counts.  See obs_diag for more.
+
+obs_assim_count/obs_assim_count.f90
+   Prints out a quick table of obs types and counts, overall start and stop times, 
+   and metadata strings and counts.  This is an older version of obs_utils/obs_assim_count.f90.
+
+`obs_seq_coverage/obs_seq_coverage.f90 <obs_seq_coverage/obs_seq_coverage.html>`_
+   Queries a set of observation sequence files to determine which observation locations report
+   frequently enough to be useful for a verification study.
+ 
+obs_total_error/obs_total_error.f90
+   Prints the total error in the mean and spread from an obs_seq file 
+   which has been through both perfect_model_obs and filter, so it has copies
+   'truth', 'ensemble mean', and 'ensemble spread'.
+   You can get more information by running the obs_diag program.
+
+Changing Observation Sequence Files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`obs_keep_a_few/obs_keep_a_few.f90 <obs_keep_a_few/obs_keep_a_few.html>`_
+   Creates an output observation sequence file that is shorter than the input obs_seq file.
+ 
+`obs_selection/obs_selection.f90 <obs_selection/obs_selection.html>`_
+   Extracts observations out of one or more obs_sequence files
+   according to a  list of observation types, times, and locations.
+   The list is usually created by :doc:`obs_seq_coverage/obs_seq_coverage`, 
+   but can be an observation sequence file,
+ 
+`>> obs_sequence_tool/obs_sequence_tool.f90 <obs_sequence_tool/obs_sequence_tool.html>`_
+   Subsets, combines, or alters observations from one or more observation sequence files 
+   and optionally writes them into a single output obs_seq file.
+
+`obs_loop/obs_loop.f90 <obs_loop/obs_loop.html>`_
+   A template to read in observations from one obs_seq file and write them,
+   optionally modified by user supplied code, to another obs_seq file.
+ 
+obs_utils/obs_sort.f90
+   Do a complete sort of an obs_seq file by location, observation type, then variance.
+   An ancestor of obs_remove_dups.
+
+obs_utils/obs_remove_dups.f90
+   Removes duplicate observations from an obs_seq file, which involves a complete sort
+   by time, location, observation type, then variance.
+ 
+`>> obs_common_subset/obs_common_subset.f90 <obs_common_subset/obs_common_subset.html>`_
+   Select the subset of observations, which were successfully assimilated, 
+   from two or more assimilation cases (which used the same obs_seq.out file).
+ 
+`obs_seq_verify/obs_seq_verify.f90 <obs_seq_verify/obs_seq_verify.html>`_
+   Reorders the observations from a forecast run of DART into a structure 
+   that is amenable for the evaluation of the forecast.
+ 
+
+obs_utils/obs_data_denial.f90
+   THIS IS NOT YET DONE!
+   Help implement a data-denial experiment by randomly changing the error variance
+   of N of each obs type in an observation sequence file to a huge value.
+ 
+Assimilation Programs
```

Comment:
i think i'd start this doc with these assimilation programs - they're the core of the project.  stuff in the current first section are not even all needed and it may send people off into looking at stuff that's not necessary.

Reply:
I organized the sections roughly by the order in which a complete, from scratch,
assimilation would be developed.  It's also a fine choice to put the core section first
but the page would be a bit more disjointed.  A flow chart might help, but maybe it doesn't matter.

Reply:
i think that's a hard one to predict - the time order that programs might be needed.  if you're a new user planning on running CAM-FV or POP, for example, you have no need of model_mod_check and sending you there first seems confusing.  if you're porting over a new model then sure, that's where you'd start - after you'd written a model_mod.f90.  

i guess i see this as a reference/overview page, not a tutorial or walkthrough, so i'm thinking "most important" first, whatever that is.

Reply:
I've moved model_mod_check down, which makes preproc the first program in the Setting Up section.
I also added a note that people may need nothing else from that section, and  a link to 
Assimilation Programs.